lib/elixir/pages/Guards.md

```diff
@@ -0,0 +1,226 @@
+# Guards
+
+Guards are a way to augment pattern matching in a predefined set of constructs.
+
+Let's see an example of a guard used in a function clause:
+
+
+```elixir
+def long_list?(list) when length(list) > 3,  do: true
+def long_list?(list) when length(list) <= 3, do: true
+```
+
+Guards start with the `when` keyword, which is followed by a boolean expression (we will define the grammar of guards more formally later on).
+
+Writing the `long_list?/1` function by only using pattern matching would be tricky: we would need to pattern match on lists of all possible lengths up to `3` and classify those as "not long" lists, and all the rest would be long lists. In code:
+
+```elixir
+def long_list?([]),               do: false
+def long_list?([_]),              do: false
+def long_list?([_, _]),           do: false
+def long_list?([_, _, _]),        do: false
+def long_list?([_, _, _, _ | _]), do: true
+```
+
+This is verbose and not scalable with the length of "long lists". Guards solve this by letting us performs additional and more complex checks than just pattern matching, allowing us to simplify clauses or even express something that is not possible to express through pattern matching.
+
+## Where guards can be used
+
+In the example above, we show how guards can be used in function clauses. In total, there are **five** constructs that allow guards:
+
+  * function clauses:
+
+  ```elixir
+  def foo(x) when is_integer(x), do: x
+  def foo(x) when is_float(x), do: round(x)
+  ```
+
+  * `case` expressions:
```

Comment:
We are also missing `match?/2`. I am honestly not convinced on the need to list all of the cases here?